Sources/SwiftKuery/Table.swift

```diff
@@ -16,9 +16,32 @@
 
 // MARK: Table
 
-/// Subclasses of the Table class are metadata describing a table in a relational database that you want to work with.
+/**
+ The `Table` class is used to represent a specific table from an SQL database in swift.
+ This class details the table name, contains the `Column` instances present inside the table and provides methods to build SQL String descriptions for the various database engines.
+ ### Usage Example: ###
+ In this example, a ToDo table class matching a table stored in an SQL database is defined.
+ The "ToDoTable" class contains the table name and three instances of the `Column` class.
+ An instance of this "ToDoTable" class is then initialized.
+ This instance is then referenced for a SQL select query to retrieve all data from a database.
+ ```swift
+ public class ToDoTable : Table {
+    let tableName = "toDoTable"
+    let toDo_id = Column("toDo_id", Int32.self, autoIncrement: true, primaryKey: true, notNull: true, unique: true)
+    let toDo_title = Column("toDo_title", String.self, notNull: true)
+    let toDo_completed = Column("toDo_completed", Bool.self, defaultValue: false)
+ }
+ 
+ public class Application {
+    let todotable = ToDoTable()
+    let selectQuery = Select(from :todotable)
```

Comment:
I get the compilation warning "Cannot use instance member 'todotable' within property initializer; property initializers run before 'self' is available" here.